### PR TITLE
xbutil2 report bug fix after testing with aie driver changes

### PR DESCRIPTION
**Pull request by Himanshu-xilinx**
* Updated intermediate schema which has changes it doesn't affect output.
* Previous changes were pushed base on spec given by aie driver but when actual aie driver changes came, they differ from spec.
* All capital letter are changed to small letter
* "dma mode" and  "dma lock id" are removed

### CODE REVIEW

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good. Thanks for making these changes

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 1]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
     tile.put("row", pt.get<uint32_t>("row"));
 
     std::string status;
-    for (auto& node: pt.get_child("core.Status", empty_pt)) {
+    for (auto& node: pt.get_child("core.status", empty_pt)) {
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:203]: Note: This comment applied to ALL of the schema name changes.

What the name change?

I ask for the last time we chatted regarding the names, it was mentioned that these names followed the documentation and they could not be changed.  Has the documentation changed?

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 2]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
...
 
-    for (auto& node: pt.get_child("core.LR", empty_pt))
+    for (auto& node: pt.get_child("core.lr", empty_pt))
       tile.put("core.link_register", node.second.data());
 
-    for (auto& node: pt.get_child("core.SP", empty_pt))
+    for (auto& node: pt.get_child("core.sp", empty_pt))
       tile.put("core.stack_pointer", node.second.data());
-
+/*
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:217]: If this code is not needed it should be removed and NOT commented out.

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 3]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
...
+*/
     boost::property_tree::ptree mm2s_array;
-    auto queue_size = pt.get_child("dma.Queue_size.MM2S").begin();
-    auto queue_status = pt.get_child("dma.Queue_status.MM2S").begin();
-    auto current_bd = pt.get_child("dma.Current_BD.MM2S").begin();
-    auto lock_id = pt.get_child("dma.Lock_ID.MM2S").begin();
+    auto queue_size = pt.get_child("dma.queue_size.mm2s").begin();
+    auto queue_status = pt.get_child("dma.queue_status.mm2s").begin();
+    auto current_bd = pt.get_child("dma.current_bd.mm2s").begin();
+    //auto lock_id = pt.get_child("dma.lock_id.mm2s").begin();
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:238]: Remove code

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 4]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
...
+    for (auto& node : pt.get_child("dma.channel_status.mm2s")) {
       boost::property_tree::ptree channel;
-      channel.put("id", id);
+      channel.put("id", id++);
       channel.put("channel_status", node.second.data());
       channel.put("queue_size", queue_size->second.data());
       channel.put("queue_status", queue_status->second.data());
       channel.put("current_bd", current_bd->second.data());
-      channel.put("lock_id", lock_id->second.data());
+      //channel.put("lock_id", lock_id->second.data());
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:247]: Remove comment

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 5]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
...
       queue_size++;
       queue_status++;
       current_bd++;
-      lock_id++;
+      //lock_id++;
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:251]: Remove comment

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 6]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
...
 
     boost::property_tree::ptree s2mm_array;
-    queue_size = pt.get_child("dma.Queue_size.S2MM").begin();
-    queue_status = pt.get_child("dma.Queue_status.S2MM").begin();
-    current_bd = pt.get_child("dma.Current_BD.S2MM").begin();
-    lock_id = pt.get_child("dma.Lock_ID.S2MM").begin();
+    queue_size = pt.get_child("dma.queue_size.s2mm").begin();
+    queue_status = pt.get_child("dma.queue_status.s2mm").begin();
+    current_bd = pt.get_child("dma.current_bd.s2mm").begin();
+    //lock_id = pt.get_child("dma.lock_id.s2mm").begin();
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:260]: remove comment

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 7]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
...
+    for (auto& node : pt.get_child("dma.channel_status.s2mm")) {
       boost::property_tree::ptree channel;
-      channel.put("id", id);
+      channel.put("id", id++);
       channel.put("channel_status", node.second.data());
       channel.put("queue_size", queue_size->second.data());
       channel.put("queue_status", queue_status->second.data());
       channel.put("current_bd", current_bd->second.data());
-      channel.put("lock_id", lock_id->second.data());
+      //channel.put("lock_id", lock_id->second.data());
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:269]: remove comment

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 8]
```diff
@@ -200,77 +200,77 @@ populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& t
...
       queue_size++;
       queue_status++;
       current_bd++;
-      lock_id++;
+      //lock_id++;
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:273]: remove comment

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 9]
```diff
@@ -589,7 +590,7 @@ ReportAie::writeReport(const xrt_core::device * _pDevice,
         if(tile.second.find("dma") != tile.second.not_found()) {
           _output << boost::format("    %s:\n") % "DMA";
           _output << boost::format("        %s:\n") % "MM2S";
-          _output << fmt12("%s") % "Mode" % tile.second.get<std::string>("dma.mm2s.mode");
+          //_output << fmt12("%s") % "Mode" % tile.second.get<std::string>("dma.mm2s.mode");
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:593]: remove comment

File: src/runtime_src/core/tools/common/ReportAie.cpp [hunk 10]
```diff
@@ -598,12 +599,12 @@ ReportAie::writeReport(const xrt_core::device * _pDevice,
             _output << fmt16("%s") % "Queue Size" % node.second.get<std::string>("queue_size");
             _output << fmt16("%s") % "Queue Status" % node.second.get<std::string>("queue_status");
             _output << fmt16("%s") % "Current BD" % node.second.get<std::string>("current_bd");
-            _output << fmt16("%s") % "Lock ID" % node.second.get<std::string>("lock_id");
+            //_output << fmt16("%s") % "Lock ID" % node.second.get<std::string>("lock_id");
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAie.cpp:602]: remove comment

File: src/runtime_src/core/tools/common/ReportAieShim.cpp [hunk 11]
```diff
@@ -177,6 +177,7 @@ populate_aie_shim(const xrt_core::device * device, const std::string& desc, boos
 
       std::string mode;
       boost::property_tree::ptree empty_pt;
+/*
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAieShim.cpp:180]: Remove commented out code

File: src/runtime_src/core/tools/common/ReportAieShim.cpp [hunk 12]
```diff
@@ -191,52 +192,57 @@ populate_aie_shim(const xrt_core::device * device, const std::string& desc, boos
...
+          channel.put("id", id++);
+          channel.put("channel_status", node.second.data());
+          channel.put("queue_size", queue_size->second.data());
+          channel.put("queue_status", queue_status->second.data());
+          channel.put("current_bd", current_bd->second.data());
+          //channel.put("lock_id", lock_id->second.data());
+          queue_size++;
+          queue_status++;
+          current_bd++;
+          //lock_id++;
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAieShim.cpp:215]: remove comment

File: src/runtime_src/core/tools/common/ReportAieShim.cpp [hunk 13]
```diff
@@ -191,52 +192,57 @@ populate_aie_shim(const xrt_core::device * device, const std::string& desc, boos
...
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAieShim.cpp:237]: remove comment

File: src/runtime_src/core/tools/common/ReportAieShim.cpp [hunk 14]
```diff
@@ -305,7 +313,7 @@ ReportAieShim::writeReport(const xrt_core::device * _pDevice,
       if(tile.second.find("dma") != tile.second.not_found()) {
         _output << boost::format("    %s:\n") % "DMA";
         _output << boost::format("        %s:\n") % "MM2S";
-        _output << fmt12("%s") % "Mode" % tile.second.get<std::string>("dma.mm2s.mode");
+        //_output << fmt12("%s") % "Mode" % tile.second.get<std::string>("dma.mm2s.mode");
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAieShim.cpp:316]: remove comment

File: src/runtime_src/core/tools/common/ReportAieShim.cpp [hunk 15]
```diff
@@ -314,12 +322,12 @@ ReportAieShim::writeReport(const xrt_core::device * _pDevice,
...
           _output << fmt16("%s") % "Queue Status" % node.second.get<std::string>("queue_status");
           _output << fmt16("%s") % "Current BD" % node.second.get<std::string>("current_bd");
-          _output << fmt16("%s") % "Lock ID" % node.second.get<std::string>("lock_id");
+          //_output << fmt16("%s") % "Lock ID" % node.second.get<std::string>("lock_id");
           _output << std::endl;
         }
 
         _output << boost::format("        %s:\n") % "S2MM";
-        _output << fmt12("%s") % "Mode" % tile.second.get<std::string>("dma.s2mm.mode");
+        //_output << fmt12("%s") % "Mode" % tile.second.get<std::string>("dma.s2mm.mode");
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAieShim.cpp:330]: remove comment

File: src/runtime_src/core/tools/common/ReportAieShim.cpp [hunk 16]
```diff
@@ -328,7 +336,7 @@ ReportAieShim::writeReport(const xrt_core::device * _pDevice,
           _output << fmt16("%s") % "Queue Size" % node.second.get<std::string>("queue_size");
           _output << fmt16("%s") % "Queue Status" % node.second.get<std::string>("queue_status");
           _output << fmt16("%s") % "Current BD" % node.second.get<std::string>("current_bd");
-          _output << fmt16("%s") % "Lock ID" % node.second.get<std::string>("lock_id");
+          //_output << fmt16("%s") % "Lock ID" % node.second.get<std::string>("lock_id");
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAieShim.cpp:339]: remove comment

File: src/runtime_src/core/tools/common/ReportAieShim.cpp [hunk 17]
```diff
@@ -191,52 +192,57 @@ populate_aie_shim(const xrt_core::device * device, const std::string& desc, boos
...
-        queue_size++;
-        queue_status++;
-        current_bd++;
-        lock_id++;
-        s2mm_array.push_back(std::make_pair("", channel));
+        boost::property_tree::ptree s2mm_array;
+        queue_size = oshim.get_child("dma.queue_size.s2mm").begin();
+        queue_status = oshim.get_child("dma.queue_status.s2mm").begin();
+        current_bd = oshim.get_child("dma.current_bd.s2mm").begin();
+        //lock_id = oshim.get_child("dma.lock_id.s2mm").begin();
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportAieShim.cpp:224]: remove comment

**Review by rozumx — CHANGES_REQUESTED** [review]
Himanshu,   There has been several changes made here where the description doesn't reflect it.

After addressing my comments, please do update the description that contains:
1. A list of the changes performed in this PR.  For example a list of "work done"
2. Since the schema changed, it needed to be at least documented in the PR.

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 1 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp:186]: this is intermediate schema which has changes it doesn't affect output.
we pushed changes base on spec given by aie driver but when actual aie driver changes came, they differ from spec.

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 2 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: according to aie driver spec these nodes were that but in actual implementation they remove but these changes will come in future. so commented this part of code. 
But now i am removeing.

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 3 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: according to aie driver spec these nodes were that but in actual implementation they remove but these changes will come in future. so commented this part of code.
But now i am removing.

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 4 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 5 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAieShim.cpp (hunk 13 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAieShim.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 6 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 7 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 8 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 10 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAie.cpp (hunk 9 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAie.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAieShim.cpp (hunk 11 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAieShim.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAieShim.cpp (hunk 12 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAieShim.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAieShim.cpp (hunk 15 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAieShim.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAieShim.cpp (hunk 14 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAieShim.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAieShim.cpp (hunk 17 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAieShim.cpp]: same as above

File: src/runtime_src/core/tools/common/ReportAieShim.cpp (hunk 16 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/common/ReportAieShim.cpp]: same as above

**Comment by Himanshu-xilinx** [comment]
Hi @rozumx 
I updates all comments and description. Please review.

**Review by rozumx — APPROVED** [review]
Thank you for making these changes.

**Pull request merged**